webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -96,46 +100,49 @@ export class ContactsEffects {
       });
   }
 
-  private loadChildren(userFacilityId) {
-    const getChildPlaces = userFacilityId !== this.selectedContact?._id;
+  private verifySelectedContactNotChanged(id) {
+    return this.selectedContact?._id !== id ? Promise.reject({code: 409}) : Promise.resolve();
+  }
+
+  private loadChildren(contactId, userFacilityId) {
+    const getChildPlaces = userFacilityId !== contactId;
     return this.contactViewModelGeneratorService
-      .loadChildren(this.selectedContact, { getChildPlaces })
-      .then(children => {
-        this.contactsActions.receiveSelectedContactChildren(children);
-      });
+      .loadChildren(this.selectedContact, {getChildPlaces})
+      .then(children => this.verifySelectedContactNotChanged(contactId)
+        .then(() => this.contactsActions.receiveSelectedContactChildren(children)));
```

Comment:
Please indent this like: 
```suggestion
      .then(children => {
        return this
          .verifySelectedContactNotChanged(contactId)
          .then(() => this.contactsActions.receiveSelectedContactChildren(children));
      });
```
Currently this kind of requires counting the number on closing brackets to see what is chained where.